client/components/repository/index.vue

```diff
@@ -53,24 +78,14 @@ export default {
       return this.getActiveUsers('repository', this.repositoryId);
     },
     tabs() {
+      const { isAdmin, isRepositoryAdmin, hasWorkflow } = this;
       const hasActivities = get(this.activities, 'length');
       const activityId = get(this.lastSelectedActivity, 'id');
-      const query = { ...this.$route.query, activityId };
-      const items = [
-        { name: 'Structure', route: 'repository', icon: 'file-tree' },
-        hasActivities && this.hasWorkflow && {
-          name: 'Progress',
-          route: 'progress',
-          icon: 'chart-timeline-variant'
-        },
-        hasActivities && { name: 'History', route: 'revisions', icon: 'history' },
-        (this.isAdmin || this.isRepositoryAdmin) && {
-          name: 'Settings',
-          route: 'repository-info',
-          icon: 'settings-outline'
-        }
-      ];
-      return items
+      const query = {
+        ...this.$route.query,
+        ...activityId && { activityId }
+      };
+      return getTabItems({ isAdmin, isRepositoryAdmin, hasWorkflow, hasActivities })
```

Comment:
Does it make sense to filter out nulls immediately in the `getTabItems` since we're already passing conditions to it?

Reply:
It does.

Reply:
👍🏻